mobius/src/cline_arc.rs

```diff
@@ -54,117 +78,137 @@ pub struct ClineArc {
+    let adjusted_c = adjusted_b + delta_c;
+    ArcAngles::new(a, adjusted_b, adjusted_c)
+}
+
+fn compute_cw_angles(a: f64, b: f64, c: f64) -> Result<ArcAngles, ArcAnglesParseError> {
+    // Compute angles in the CW direction
+    let delta_b = (a - b).rem_euclid(TAU);
+    let delta_c = (b - c).rem_euclid(TAU);
+
+    let adjusted_b = a - delta_b;
+    let adjusted_c = adjusted_b - delta_c;
+    ArcAngles::new(a, adjusted_b, adjusted_c)
+}
+
 impl ClineArc {
-    pub fn classify(&self) -> ClineArcGeometry {
+    fn compute_line_geometry(&self) -> ClineArcGeometry {
+        if let Complex::Infinity = self.a {
+            // ray goes inf -> b -> c
+            // but it looks like inf <- c
+            let to_infinity = (self.b - self.c).normalize().unwrap();
+            return ClineArcGeometry::FromInfinity(Ray {
+                start: self.c,
+                unit_dir: to_infinity,
+            });
+        }
+
+        if let Complex::Infinity = self.c {
+            // ray goes a -> b -> inf
+            let to_infinity = (self.b - self.a).normalize().unwrap();
+            return ClineArcGeometry::ToInfinity(Ray {
+                start: self.a,
+                unit_dir: to_infinity,
+            });
+        }
+
+        if let Complex::Infinity = self.b {
+            // ray goes    inf <- a    c -> inf
+            let ac = (self.c - self.a).normalize().unwrap();
+            return ClineArcGeometry::ThruInfinity(DoubleRay(
+                Ray {
+                    start: self.a,
+                    unit_dir: -ac,
+                },
+                Ray {
+                    start: self.c,
+                    unit_dir: ac,
+                },
+            ));
+        }
+
+        // All three points are finite so now we we need to check if
+        // if they're in order a -> b -> c. If not, then
+        // the line goes through infinity like
+        //   inf <- a   c <- b <- inf
+        //   inf <- b <- a  c -> inf
+        // in which case we just want a ray pair through a and c.
+        //
+        // we can just check a -> b and b -> c. If the dot product is
+        // positive, then the points are in order.
+        let in_order = Complex::dot(self.b - self.a, self.c - self.b) > 0.0;
+
+        if in_order {
+            ClineArcGeometry::LineSegment(LineSegment {
+                start: self.a,
+                end: self.c,
+            })
+        } else {
+            let ac = (self.c - self.a).normalize().unwrap();
+            ClineArcGeometry::ThruInfinity(DoubleRay(
+                Ray {
+                    start: self.a,
+                    unit_dir: -ac,
+                },
+                Ray {
+                    start: self.c,
+                    unit_dir: ac,
+                },
+            ))
+        }
+    }
+
+    fn compute_circle_geometry(&self, circle: Circle) -> Result<ClineArcGeometry, ClineArcError> {
+        // Determine if the 3 points circulate counterclockwise or
```

Comment:
This is one case where you do want to use 3 angles to compute the angle

Reply:
Though... Should this be moved to `CircularArc?` you're taking a circle and 3 points on that circle and computing a `CircularArc`. sounds like a constructor?

Reply:
let's see if anything else needs this first.